src/rust/engine/process_execution/src/docker_tests.rs

```diff
@@ -735,8 +758,21 @@ async fn run_command_via_docker_in_dir(
     store.unwrap_or_else(|| Store::local_only(executor.clone(), store_dir.path()).unwrap());
   let (_caches_dir, named_caches, immutable_inputs) =
     named_caches_and_immutable_inputs(store.clone());
-  pull_image(docker, IMAGE).await;
+  let local_runner = local::CommandRunner::new(
+    store.clone(),
+    executor.clone(),
+    dir.clone(),
+    named_caches.clone(),
+    immutable_inputs.clone(),
+    cleanup,
+  );
+
+  if let Some(docker) = docker_opt {
+    pull_image(docker, IMAGE).await;
```

Comment:
@tdyas: I missed this before. Is this something that we expect to do automatically in the future when starting a cached image?

Reply:
We need to give some thought here. It will probably need to be configurable for, among other reasons, security purposes so that users have control when images updated. (Recall that tags in Docker are not immutable.) See https://github.com/pantsbuild/pants/issues/16767 for discussion. Both Docker and Kubernetes make this configurable (discussion in the issue).